### PR TITLE
Remove unecessary compat macros and fix deprecation warnings

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

**Comment by musm** [comment]
Build failures are unrelated. 
0.4 failures related to Cairo not building.
nightly os x, i have seen issues with it not building hdf5 correctly so I also assume these are unrelated.

**Comment by musm** [comment]
ok retriggering CI after updated appveyor and travis

**Comment by musm** [comment]
does anyone understand the 0.4 failures?

**Comment by tkelman** [comment]
Where you've replaced `takebuf_string` is inside macro-output code. So you can't assume Compat has been imported at the macro expansion location, it needs to work without Compat.

**Comment by musm** [comment]
Nice one. Thank you. I will try to think of a way around this

File: src/dependencies.jl [hunk 1]
```diff
@@ -884,7 +884,11 @@ macro install(_libmaps...)
                         println(depsfile_buffer)
                         println(depsfile_buffer, "# Load-hooks")
                         println(depsfile_buffer, join(load_hooks,"\n"))
-                        depsfile_content = chomp(String(take!(depsfile_buffer)))
+                        if VERSION < v"0.5-" # remove this case after dropping 0.4 support
```
Comment (tkelman) [inline src/dependencies.jl]: `take!` on a buffer is much more recent than this, and need to be specific with VERSION cutoffs

File: src/dependencies.jl [hunk 2]
```diff
@@ -884,7 +884,11 @@ macro install(_libmaps...)
                         println(depsfile_buffer)
                         println(depsfile_buffer, "# Load-hooks")
                         println(depsfile_buffer, join(load_hooks,"\n"))
-                        depsfile_content = chomp(takebuf_string(depsfile_buffer))
+                        if VERSION < v"0.6.0-dev.1256"
+                            depsfile_content = chomp(String(takebuf_array(depsfile_buffer)))
```
Comment (tkelman) [inline src/dependencies.jl]: `takebuf_string` existed on 0.4, but String had a different meaning there

Comment (musm) [inline src/dependencies.jl]: a right, sorry I keep messing this up.....

Comment (tkelman) [inline src/dependencies.jl]: it's alright, thanks for helping out

File: src/dependencies.jl [hunk 3]
```diff
@@ -884,9 +884,9 @@ macro install(_libmaps...)
                         println(depsfile_buffer)
                         println(depsfile_buffer, "# Load-hooks")
                         println(depsfile_buffer, join(load_hooks,"\n"))
-                        if VERSION < v"0.6.0-dev.1256"
-                            depsfile_content = chomp(String(takebuf_array(depsfile_buffer)))
-                        else
+                        if v"0.4" < VERSION < v"0.6.0-dev.1256"
```
Comment (tkelman) [inline src/dependencies.jl]: no need for the lower bound, that's taken care of by the package REQUIRE

File: src/dependencies.jl [hunk 4]
```diff
@@ -884,9 +884,9 @@ macro install(_libmaps...)
...
+                            depsfile_content = chomp(takebuf_string(depsfile_buffer))
+			else
```
Comment (tkelman) [inline src/dependencies.jl]: spaces please

**Comment by musm** [comment]
yeah the web interface is awful for making minor changes...

File: REQUIRE [hunk 5]
```diff
@@ -1,4 +1,4 @@
 julia 0.4
 URIParser
 SHA
-Compat 0.8.4
+Compat 0.9.5
```
Comment (tkelman) [inline REQUIRE]: by making the returned code version-dependent we've removed the need to bump the compat minimum

**Comment by tkelman** [comment]
nightly failures are ColorTypes having issues with the type system rewrite

**Pull request merged**